hal/mbed_mpu_api.c

```diff
@@ -0,0 +1,120 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "hal/mpu_api.h"
+#include "cmsis.h"
```

Comment:
Since we are including cmsis.h should we directly access MPU registers or is it better to use the wrapping functions such as in mpu_armv7.h

Reply:
I updated this to use some mpu_armv7.h macros but left some out as they added more indirection with no increase in readability.